modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/_diffs/plugins/toolbarbuttons/plugin.js

```diff
@@ -50,22 +225,75 @@
 			});
 
 			editor.ui.addRichCombo('TableHeaders', {
+				_headersLabels: {
+					[HEADING_BOTH]: `${editor.lang.table.headers}: ${editor.lang.table.headersBoth}`,
+					[HEADING_COL]: `${editor.lang.table.headers}: ${editor.lang.table.headersColumn}`,
+					[HEADING_NONE]: `${editor.lang.table.headers}: ${editor.lang.table.headersNone}`,
+					[HEADING_ROW]: `${editor.lang.table.headers}: ${editor.lang.table.headersRow}`,
+				},
+
+				_setHeading(heading, editor) {
+					setHeading({editor, heading, table: null});
+
+					this.setValue(heading, this._headersLabels[heading]);
+				},
+
 				init() {
-					const headersPrefix = editor.lang.table.headers;
+					this.add(
+						HEADING_NONE,
+						this._headersLabels[HEADING_NONE],
+						this._headersLabels[HEADING_NONE]
+					);
+					this.add(
+						HEADING_ROW,
+						this._headersLabels[HEADING_ROW],
+						this._headersLabels[HEADING_ROW]
+					);
+					this.add(
+						HEADING_COL,
+						this._headersLabels[HEADING_COL],
+						this._headersLabels[HEADING_COL]
+					);
+					this.add(
+						HEADING_BOTH,
+						this._headersLabels[HEADING_BOTH],
+						this._headersLabels[HEADING_BOTH]
+					);
 
-					const headersNone = `${headersPrefix}: ${editor.lang.table.headersNone}`;
-					const headersRow = `${headersPrefix}: ${editor.lang.table.headersRow}`;
-					const headersColumn = `${headersPrefix}: ${editor.lang.table.headersColumn}`;
-					const headersBoth = `${headersPrefix}: ${editor.lang.table.headersBoth}`;
+					const currentHeading = getHeading({editor, table: null});
```

Comment:
No need to set `table: null`, we can just omit it to have falsy value.

Reply:
Ok, changed